src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -516,7 +516,7 @@ internal SafeFileHandle OpenPort(string portName)
                 }
                 else
                 {
-                    throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+                    throw new ArgumentException(SR.Arg_InvalidSerialPortExtended, portName);
```

Comment:
I agree that is better to show the invalid port name, but, to do it properly you need to change the message to include it and use SR.Format to build the string. The nameof in this case should be kept since it indicates the name of the parameter (see the ArgumentException constructor). Same change should be made on Windows side.

Reply:
do you have specific example @pjanotti ?

Reply:
https://github.com/dotnet/corefx/blob/e0ba7aa8026280ee3571179cc06431baf1dfaaac/src/System.IO.Ports/src/System/IO/Ports/InternalResources.cs#L83

So it shoud be `throw new ArgumentException(SR.Format(SR.Arg_InvalidSerialPort, portName), nameof(portName));`

This should be symmetrical in Windows.

Reply:
That does not seems to work:

```
Support/TCSupport.cs:    PrintInfo("Exception opening port {0}: {1}", portName, e);

  Exception opening port /dev/foo: System.ArgumentException: The given port name does not start with COM/com or does not resolve to a valid serial port.
  Parameter name: portName
```

the port name is printed by test code and it is not part of the exception message.
Maybe we can leave it as that instead of getting name into exception it self. 

I purposely did not use Arg_InvalidSerialPort as I think the referral to COM prefix does not make sense on Unix.

Reply:
You need to add a '{0}' to the SR.Arg_InvalidSerialPort and update it to work with Windows and Unix, e.g.: `The given port name ({0}) does not resolve to a valid serial port.`

Reply:
thanks. It now throws:

>  Exception opening port /dev/foo: System.ArgumentException: The given port name (/dev/foo) does not resolve to a valid serial port.
>  Parameter name: portName